src/experimental/HomeChainCoordinator.sol

```diff
@@ -0,0 +1,363 @@
+        uint256 _index,
+        bytes calldata _psbtData,
+        bytes calldata _options,
+        address _refundAddress
+    ) external payable whenNotPaused nonReentrant onlyOwner {
+        bytes32 merkleRoot = lightClient.getMerkleRootForBlock(_blockHash);
+        _sendMessage(merkleRoot, _btcTxnHash, _proof, _index, _psbtData, _options, _refundAddress);
+    }
+
+    /**
+     * @dev Internal function to send a cross-chain message with PSBT data.
+     * @param _merkleRoot The merkle root for the block.
+     * @param _btcTxnHash The BTC transaction hash.
+     * @param _proof The proof for the transaction.
+     * @param _index The index of the transaction in the block.
+     * @param _psbtData The PSBT data to be processed.
+     * @param _options LayerZero message options.
+     * @param _refundAddress The address to refund in case of failure.
+     */
+    function _sendMessage(
+        bytes32 _merkleRoot,
+        bytes32 _btcTxnHash,
+        bytes32[] calldata _proof,
+        uint256 _index,
+        bytes calldata _psbtData,
+        bytes calldata _options,
+        address _refundAddress
+    ) internal {
+        // 0. btcTxnHash generated from the psbt data being shared should be the same as the one passed
+        bytes32 txid = TxidCalculator.calculateTxid(_psbtData);
+        if (txid != _btcTxnHash) {
+            revert InvalidPSBTData();
+        }
+
+        // 1. Parse and validate PSBT data
+        BitcoinTxnParser.TransactionMetadata memory metadata = _validatePSBTData(_psbtData);
+
+        // 2. TODO: Get _dstEid for a specific metadata.chainId from LayerZero contract
+        uint32 _dstEid = metadata.chainId == 8453 ? 30184 : metadata.chainId;
+        console2.log("Destination chain ID: ", _dstEid);
+
+        // 3. Check if the message already exists or is processed
+        if (btcTxnHash_psbtMetadata[_btcTxnHash].isMinted) {
+            revert TxnAlreadyProcessed(_btcTxnHash);
+        }
+
+        // 4. Validate receiver is set for destination chain
+        if (peers[_dstEid] == bytes32(0)) {
+            revert InvalidReceiver();
+        }
+
+        // 5. Validate txn with SPV data
+        require(BitcoinUtils.verifyTxInclusion(_btcTxnHash, _merkleRoot, _proof, _index), BitcoinTxnNotFound());
+
+        // TODO: This needs to come from the metadata itself as this will keep on changing
+        bytes32 networkPublicKey;
+        // 6. Store PSBT metadata
+        PSBTMetadata memory psbtMetaData = PSBTMetadata({
+            isMinted: true,
+            chainId: metadata.chainId,
+            user: metadata.receiverAddress,
+            lockedAmount: metadata.lockedAmount,
+            nativeTokenAmount: metadata.nativeTokenAmount,
+            networkPublicKey: networkPublicKey,
+            psbtData: _psbtData
+        });
+        btcTxnHash_psbtMetadata[_btcTxnHash] = psbtMetaData;
+
+        // 7. Send message through LayerZerobytes memory payload
+        bytes memory payload = abi.encode(
```

Comment:
do we have to add a unique identifier to add a prefix to make it unique per message otherwise it can mingle up with the other message type

Reply:
We might when the changes for the burn are implemented, but I believe the flow for the message will happen from BaseChainCoordinator to HomeChainCoordinator, so adding a type for the message might not be required.